src/jmh/java/hu/akarnokd/comparison/scrabble/ShakespearePlaysScrabbleWithCyclopsReactOpt.java

```diff
@@ -49,145 +66,99 @@
     )
     @Fork(1)
     public List<Entry<Integer, List<String>>> measureThroughput() throws InterruptedException {
-
-        //  to compute the score of a given word
-        Function<Integer, Integer> scoreOfALetter = letter -> letterScores[letter - 'a'];
-
-        // score of the same letters in a word
-        Function<Entry<Integer, MutableLong>, Integer> letterScore =
-                entry ->
-                        letterScores[entry.getKey() - 'a'] *
-                        Integer.min(
-                                (int)entry.getValue().get(),
-                                scrabbleAvailableLetters[entry.getKey() - 'a']
-                            )
-                    ;
-
-
-        Function<String, ReactiveSeq<Integer>> toIntegerIx =
-                string -> chars(string);
-
-        // Histogram of the letters in a given word
-        Function<String, ReactiveSeq<HashMap<Integer, MutableLong>>> histoOfLetters =
-                word ->  toIntegerIx.apply(word)
-                            .scanLeft(new HashMap<Integer, MutableLong>(), (map, value) -> {
-                                MutableLong newValue = map.get(value) ;
-                                if (newValue == null) {
-                                    newValue = new MutableLong();
-                                    map.put(value, newValue);
-                                }
-                                newValue.incAndSet();
-                                return map;
-                            })
-                        .takeRight(1);
-                 ;
-
-        // number of blanks for a given letter
-        Function<Entry<Integer, MutableLong>, Long> blank =
-                entry ->
-                        Long.max(
-                            0L,
-                            entry.getValue().get() -
-                            scrabbleAvailableLetters[entry.getKey() - 'a']
-                        )
-                    ;
-
+                                               
+         // Histogram of the letters in a given word
+        Function<String, ReactiveSeq<Map.Entry<Integer, Long>>> histoOfLetters =
```

Comment:
In these benchmarks, functions such as `scoreOfALetter`, `letterScore` should be kept separate as they were, adjusting their types to utilize as much `ReactiveSeq` as possible. Beyond that, yo can pick a better processing algorithm but the dataflow steps should be kept separate. The inlining is the job of the JIT after that.

In addition, you should rebase your PR as it conflicts.